src/service/search/datafusion/regexp_udf.rs

```diff
@@ -267,4 +461,69 @@ mod tests {
         assert!(is_valid_character_after_escape('d'));
         assert!(!is_valid_character_after_escape('a'));
     }
+
+    #[tokio::test]
+    async fn test_regexp_match_to_fields_udf() {
+        let log_line = r#"2024-02-29 00:15:30 15.128.22.213 GET /Administradores_Elina/service-worker.js - 443"#;
+
+        let schema = Arc::new(Schema::new(vec![Field::new("log", DataType::Utf8, false)]));
+
+        let batch = RecordBatch::try_new(
+            schema.clone(),
+            vec![Arc::new(StringArray::from(vec![log_line]))],
+        )
+        .unwrap();
+
+        let ctx = SessionContext::new();
+        let my_udf = ScalarUDF::from(RegxpMatchToFields::new());
+        ctx.register_udf(my_udf);
+
+        let provider = MemTable::try_new(schema, vec![vec![batch]]).unwrap();
+        ctx.register_table("t", Arc::new(provider)).unwrap();
+
+        let sqls = vec![
+            (
+                r#"select regexp_match_to_fields(log, '(?P<timestamp>[^\s]+ [^\s]+) (?P<client_ip>[^\s]+) (?P<http_method>[^\s]+) (?P<requested_path>[^\s]+) (?P<placeholder1>[^\s]+) (?P<server_port>[^\s]+)') as subquery from t"#,
+                vec![
+                    "+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+",
+                    "| subquery                                                                                                                                                                  |",
+                    "+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+",
+                    "| {timestamp: 2024-02-29 00:15:30, client_ip: 15.128.22.213, http_method: GET, requested_path: /Administradores_Elina/service-worker.js, placeholder1: -, server_port: 443} |",
+                    "+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+",
+                ],
+            ),
+            (
+                r#"select regexp_match_to_fields(log, '([^\s]+ [^\s]+) ([^\s]+) ([^\s]+)') as subquery from t"#,
+                vec![
+                    "+-----------------------------------------------------------------------------+",
+                    "| subquery                                                                    |",
+                    "+-----------------------------------------------------------------------------+",
+                    r#"| {([^\s]+ [^\s]+: 2024-02-29 00:15:30, ([^\s]+: 15.128.22.213, ([^\s]+: GET} |"#,
```

Comment:
Returning such results seems not very understandable. How about we add some restrictions? If there is no name in the regular expression, return an error. This function requires a named regular expression.

Reply:
Updated to require the use of named capturing groups.